compiler/semcall.nim

```diff
@@ -253,6 +255,8 @@ proc presentFailedCandidates(c: PContext, n: PNode, errors: CandidateErrors):
       candidates.add(diag & "\n")
     candidatesAll.add candidates
   candidatesAll.sort # fix #13538
+  if immutParam.len > 0:
+    candidatesAll.insert(("\nAtleast one mismatch is due to " & immutParam & " being immutable\n\n").quoteExpr)
```

Comment:
good idea; I'd go even further and only show the procs with mismatch due to mutable/immutable when there's at least one of them, and rely on `--showAllMismatches:on` for showing all, but that would definitely belong to future PR